OpenRA.Mods.Common/Traits/World/ColorPickerManager.cs

```diff
@@ -91,31 +84,30 @@ Color MakeValid(float hue, float sat, float val, MersenneTwister random, IEnumer
 			sat = sat.Clamp(HsvSaturationRange[0], HsvSaturationRange[1]);
 			val = val.Clamp(HsvValueRange[0], HsvValueRange[1]);
 
-			// Limit to 100 attempts, which is enough to move all the way around the hue range
-			string errorMessage = null;
-			var stepSign = 0;
-			for (var i = 0; i < 101; i++)
+			string errorMessage;
+			var color = Color.FromAhsv(hue, sat, val);
+			if (IsInvalidColor(color, terrainColors))
+				errorMessage = PlayerColorTerrain;
+			else if (IsInvalidColor(color, playerColors))
+				errorMessage = PlayerColorPlayer;
+			else
+				return color;
+
+			// Move by expanding from the selected color in both directions by a limited amount circling
+			// around the hue a bunch of times. This method usually returns a color similar the selected
+			// and controlls the randomness.
+			for (var i = 2; i < 401; i++)
```

Comment:
Why 401?

Reply:
400 times, no specific reason for why other than the loop must have an end

Reply:
added a comment